allennlp/data/iterators/bucket_iterator.py

```diff
@@ -97,7 +65,7 @@ class BucketIterator(DataIterator):
 
     def __init__(
         self,
-        sorting_keys: List[Tuple[str, str]],
+        sorting_keys: List[Tuple[str, str]] = None,
```

Comment:
FYI this now means that the `BasicIterator` can be deprecated and just subclassed with this.

Reply:
Yes, agreed, we can remove the BasicIterator entirely for 1.0.

Reply:
Well, we might want to add an option to this one to skip sorting, just in case you have data that really doesn't need it and don't want to waste computation with a useless sort.

Reply:
Yes, I think we should retain an option to not sort. (Whether it's a separate iterator or a flag.) This is a bit of an edge case, but when debugging perf issues the bucket iterator really complicated understanding what was going on by introducing long periods of sorting. I suspect there are other cases where not sorting will be desired -- perhaps when your model is very sensitive to input lengths. (One could increase padding_noise excessively, but then we'll just be wasting compute sorting the noise.)